### PR TITLE
Implement SoftMax and NLLLoss double backwards.

### PR DESCRIPTION
**Pull request by gchanan**
Also simplify the SoftMax documentation to give the straightforward formula rather than the numerically stable formula, which seems like an implementation detail.

### CODE REVIEW

**Comment by soumith** [comment]
Doesn't CrossEntropyCriterion use LogSoftmax + NLLLoss (not SoftMax + NLLLoss)

**Comment by gchanan** [comment]
Yes.  This could be two separate PRs; there is no meaning for them being together beyond that's the order I worked on them and it was easier to submit them together.

**Comment by soumith** [comment]
ah okay. got it.

**Comment by gchanan** [comment]
reopening this now that ATen bug has been fixed; let's see if this passes.
